internal/hash.h

```diff
@@ -20,6 +20,7 @@ struct ar_table_struct;
 typedef unsigned char ar_hint_t;
 
 enum ruby_rhash_flags {
+    RHASH_COPY_ON_WRITE = FL_USER0,
```

Comment:
String calls this flag `STR_SHARED` and array `RARRAY_SHARED_FLAG`. It would be nice if this was consistent and called `RHASH_SHARED`.